### ISSUE TITLE
Intersecting multipolygon ways is not detected as an error

### ISSUE THREAD

**Issue by fqdb** [openstreetmap/iD#7766]
This rendering error is not identified as an error in iD (I tried both the live version hosted on openstreetmap.org [2.17.3] and the preview of iD 3):

https://wiki.openstreetmap.org/wiki/Relation:multipolygon/validity#2_-_Overlapping.2C_unclosed_member_ways_belonging_to_the_same_role

![Annotation 2020-07-04 174750](https://user-images.githubusercontent.com/2826336/86516178-b4cf4400-be1e-11ea-91eb-ceae524afaa0.jpg)

Upon uploading the data, it won't be rendered by OSM Carto.

My system is:
```
Opera Browser 
Version:68.0.3618.173
Update stream:Stable
System:Windows 10 64-bit

```

**Comment by quincylvania** [comment]
@fqdb Could you link to a location where this pattern occurs? It's unclear to me what's going on in the screenshot. It doesn't look like the example you linked to on the wiki since none of the ways appear to be unclosed, but I'm not sure.

**Comment by fqdb** [comment]
> @fqdb Could you link to a location where this pattern occurs? It's unclear to me what's going on in the screenshot. It doesn't look like the example you linked to on the wiki since none of the ways appear to be unclosed, but I'm not sure.

The place where I originally found wrong data like this was already fixed, but if you draw a new object such as shown in the picture it won't be detected as an error.

To clarify what the error is: multipolygon ways may not share a set of consecutive points. If it would be uploaded like that, the renderer will not show the multipolygon.

**Comment by fairithTM** [comment]
I agree with @fqdb, creating invalid multipolygons like this should give at least a warning in iD.


> It's unclear to me what's going on in the screenshot.

The problem in that screenshot should be the "8"-shaped part in the middle. It consists of two rectangles which share the line in the middle. This line segment being part of the multipolygon twice makes it invalid from my understanding.

A common source of this specific type of invalid multipolygons is merging ("right click"->"+" in iD) multiple adjacent buildings that each share a wall. To inexperienced mappers this may often seem like a very good idea. The resulting relation is however an invalid multipolygon. A lot of these cases might be prevented if iD would show a warning/error in such cases instead of silently uploading it.


> Could you link to a location where this pattern occurs?

https://www.openstreetmap.org/relation/966423
![grafik](https://user-images.githubusercontent.com/101255272/183957630-fe85d52b-2140-48b3-a9b3-1523a5ac667a.png)
https://www.openstreetmap.org/relation/7027423
![grafik](https://user-images.githubusercontent.com/101255272/183957705-2e120ec3-c49f-44e1-9aba-09606b9233d6.png)
https://www.openstreetmap.org/relation/13400263
![grafik](https://user-images.githubusercontent.com/101255272/183957893-6b6bbb58-180a-4bd5-926b-9373833cf0e0.png)
Countless (better?) examples can be found marked by error detection tools:
https://tools.geofabrik.de/osmi/?view=areas&lon=0.00000&lat=0.00000&zoom=3&baselayer=Geofabrik%20Standard&overlays=duplicate_segment%2Cways

**Comment by 1ec5** [comment]
I think some of the comments in #435 allude to the idea that merging two areas that touch or overlap should actually union them into a single area instead of a multipolygon. This would automatically address some of the cases where you see this issue, though it would exacerbate the problem of creating conflicting tags, which the UI should warn about: #908.
